### PR TITLE
Make KeywordArgs fail if unexpected keys are passed in

### PR DESCRIPTION
**Pull request by abevoelker**
Wasn't sure where to put the KeywordArgs specs - let me know if it should go in spec/contracts_spec.rb or whatnot (or if I'm otherwise way off base)

### CODE REVIEW

**Comment by abevoelker** [comment]
Hmm weird, specs passed with 2.2 on my machine. I'll look into the failures.

**Comment by abevoelker** [comment]
Ah `bundle exec rake` was passing, but it seems that `bundle exec rspec` does fail on my machine as it runs more specs than rake...

**Comment by alex-fedorov** [comment]
Looks like you actually want to fix this spec. Because it tries to pass in unexpected argument and expects it to not fail. Seems like a buggy behavior to me.

**Comment by abevoelker** [comment]
Okay I think I fixed it - let me know if I violated the intent of the spec or whatnot

File: lib/contracts/builtin_contracts.rb [hunk 1]
```diff
@@ -385,9 +385,10 @@ def initialize(options)
     end
 
     def valid?(hash)
-      options.all? do |key, contract|
-        Optional._valid?(hash, key, contract)
-      end
+      (hash.keys - options.keys == []) &&
```
Comment (alex-fedorov) [inline lib/contracts/builtin_contracts.rb]: Can you use guard statement here instead - current code makes it hard to read. Example:

```ruby
def valid?(hash)
  return false unless hash.keys - options.keys == []      # <--- this is a guard statement
  # .. do the main job here ..
end
```

**Comment by alex-fedorov** [comment]
Other than that, looks good.

File: lib/contracts/builtin_contracts.rb (hunk 1 above)
Comment (abevoelker) [inline lib/contracts/builtin_contracts.rb]: Oh yeah for sure, should've asked about that

**Comment by alex-fedorov** [comment]
@abevoelker Thanks!

**Pull request merged**